lib/auth/types/openid/OpenId.js

```diff
@@ -73,7 +80,72 @@ export default class OpenId extends AuthType {
         }
     }
 
-    async authenticate(credentials)  {
+    setupAuthScheme() {
+        this.server.auth.scheme('security_access_control_scheme', (server, options) => ({
+            authenticate: async (request, h) => {
+                let credentials = null;
+
+                if (this.routesToIgnore.includes(request.path)) {
+                    return h.continue;
+                }
+
+                // let configured routes that are not under our control pass,
+                // for example /api/status to check Kibana status without a logged in user
+                if (this.unauthenticatedRoutes.includes(request.path)) {
+                    credentials = this.server.plugins.opendistro_security.getSecurityBackend().getServerUser();
+                    return h.authenticated({ credentials });
+                };
+
+                try {
+                    credentials = await this.server.auth.test('security_access_control_cookie', request);
+                    return h.authenticated({ credentials })
+                } catch (error) {
+                    let authHeaderCredentials = this.detectAuthHeaderCredentials(request);
```

Comment:
Are we not doing anything for "_cookieValidationError" like in superclass?

Reply:
_cookieValidationError is removed as the code to check for the error is removed.

With respect to superclass there is a considerable change for handling the refresh_token.